File: src/main/java/org/ld/beans/ResponseBodyBean.scala
package org.ld.beans

import scala.beans.BeanProperty

/*
 * @author ld
 * ResponseBody bean 实体
 */
class ResponseBodyBean[T] {
  @BeanProperty var errorCode: Integer = _
  @BeanProperty var message: String = _
  @BeanProperty var data: T = _
  @BeanProperty var stackTrace: Array[String] = _
}

File: src/main/java/org/ld/beans/ResponseBodyBean.java
package org.ld.beans;

public class ResponseBodyBean<T> {

    private Integer errorCode ;
    private String message;
    private T data;
    private String[] stackTrace;

    public Integer getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(Integer errorCode) {
        this.errorCode = errorCode;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public String[] getStackTrace() {
        return stackTrace;
    }

    public void setStackTrace(String[] stackTrace) {
        this.stackTrace = stackTrace;
    }
}
